service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -67,9 +77,84 @@
     @Mock
     private GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
 
+    @Mock
+    private C2UploadedEmailContentProvider c2UploadedEmailContentProvider;
+
+    @Mock
+    private IdamApi idamApi;
+
+    @Mock
+    private UserDetailsService userDetailsService;
+
     @InjectMocks
     private NotificationHandler notificationHandler;
 
+    @Nested
+    class C2UploadedNotificationChecks {
+        final String subjectLine = "Lastname, SACCCCCCCC5676576567";
+
+        @Test
+        void shouldNotNotifyHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+            final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+                .put("subjectLine", subjectLine)
+                .put("hearingDetailsCallout", subjectLine)
+                .put("reference", "12345")
+                .put("caseUrl", "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
+                .build();
+
+            given(idamApi.retrieveUserDetails(AUTH_TOKEN))
+                .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                    "Hmcts", "Test", UserRole.HMCTS_ADMIN.getRoles()));
+
+            given(userDetailsService.getUserDetails(AUTH_TOKEN))
+                .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                    "Hmcts", "Test", UserRole.HMCTS_ADMIN.getRoles()));
+
+            given(c2UploadedEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
+                .willReturn(parameters);
+
+            notificationHandler.sendNotificationForC2Upload(new C2UploadedEvent(callbackRequest(), AUTH_TOKEN, USER_ID));
+
+            SendEmailResponse response = verify(notificationClient, times(0))
+                .sendEmail(eq(C2_UPLOAD_NOTIFICATION_TEMPLATE), eq("hmcts-admin@test.com"),
+                    eq(parameters), eq("12345"));
+
+            assertThat(response).isNull();
+        }
+
+        @Test
+        void shouldNotifyNonHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+            final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+                .put("subjectLine", subjectLine)
+                .put("hearingDetailsCallout", subjectLine)
+                .put("reference", "12345")
+                .put("caseUrl", "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
+                .build();
+
+            given(userDetailsService.getUserDetails(AUTH_TOKEN))
```

Comment:
I still don't understand why we mock this. All this method does is called idamApi.retrieveUserDetails. Surely one mock is all we need